viewer/src/main/webapp/viewer-html/components/Legend.js

```diff
@@ -293,8 +293,9 @@ Ext.define("viewer.components.Legend", {
         if (legendScale < mapResolution) {
             return;
         }
-        
-        // TODO when layer is out of scale we could also decide not to show the legend for this layer
+        if(curScale > legendScale){
```

Comment:
What is the difference with the statement above? That fixed the Legend scale issue for me. Do we need both checks or is one sufficient?